wrapper/src/test/java/integration/refactored/container/tests/ReadWriteSplittingTests.java

```diff
@@ -516,4 +788,242 @@ public void test_transactionResolutionUnknown() throws SQLException {
+          break;
+        }
+      }
+      if (otherReaderId.equals("")) {
+        fail("could not acquire new reader ID");
+      }
+
+      // Kill all instances except one other reader
+      for (int i = 0; i < instances.size(); i++) {
+        final String instanceId = instances.get(i).getInstanceId();
+        if (otherReaderId.equals(instanceId)) {
+          continue;
+        }
+        ProxyHelper.disableConnectivity(instanceId);
+      }
+
+      assertFirstQueryThrows(conn, SqlState.COMMUNICATION_LINK_CHANGED.getState());
+      assertFalse(conn.isClosed());
+      String currentConnectionId = queryInstanceId(conn);
+      assertEquals(otherReaderId, currentConnectionId);
+      assertNotEquals(readerConnectionId, currentConnectionId);
+
+      ProxyHelper.enableAllConnectivity();
+
+      conn.setReadOnly(false);
+      currentConnectionId = queryInstanceId(conn);
+      assertEquals(writerConnectionId, currentConnectionId);
+
+      conn.setReadOnly(true);
+      currentConnectionId = queryInstanceId(conn);
+      assertEquals(otherReaderId, currentConnectionId);
+    }
+  }
+
+  @TestTemplate
+  @EnableOnDatabaseEngine(DatabaseEngine.MYSQL)
+  @EnableOnDatabaseEngineDeployment(DatabaseEngineDeployment.AURORA)
+  public void test_failoverReaderToWriter_setReadOnlyTrueFalse() throws SQLException, IOException {
+    try (final Connection conn =
+             DriverManager.getConnection(getProxiedUrl(), this.defaultProps)) {
+
+      final String writerConnectionId = queryInstanceId(conn);
+
+      conn.setReadOnly(true);
+      final String readerConnectionId = queryInstanceId(conn);
+      assertNotEquals(writerConnectionId, readerConnectionId);
+
+      List<TestInstanceInfo> instances = TestEnvironment.getCurrent().getInfo().getDatabaseInfo().getInstances();
+
+      // Kill all instances except the writer
+      for (int i = 0; i < instances.size(); i++) {
+        final String instanceId = instances.get(i).getInstanceId();
+        if (writerConnectionId.equals(instanceId)) {
+          continue;
+        }
+        ProxyHelper.disableConnectivity(instanceId);
+      }
+
+      assertFirstQueryThrows(conn, SqlState.COMMUNICATION_LINK_CHANGED.getState());
+      assertFalse(conn.isClosed());
+      String currentConnectionId = queryInstanceId(conn);
+      assertEquals(writerConnectionId, currentConnectionId);
+
+      ProxyHelper.enableAllConnectivity();
+
+      conn.setReadOnly(true);
+      currentConnectionId = queryInstanceId(conn);
+      assertNotEquals(writerConnectionId, currentConnectionId);
+
+      conn.setReadOnly(false);
+      currentConnectionId = queryInstanceId(conn);
+      assertEquals(writerConnectionId, currentConnectionId);
+    }
+  }
+
+  @TestTemplate
+  @EnableOnDatabaseEngine(DatabaseEngine.MYSQL)
+  @EnableOnDatabaseEngineDeployment(DatabaseEngineDeployment.AURORA)
+  public void test_multiHostUrl_topologyOverridesHostList() throws SQLException {
```

Comment:
IT's confusing test. What's the purpose of it?

Reply:
I left a comment on this, I think this test is outdated and can be removed.